src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LoweredDynamicOperationFactory.cs

```diff
@@ -693,12 +693,12 @@ private static NamedTypeSymbol CreateCallSiteContainer(SyntheticBoundNodeFactory
             int generation = factory.CompilationState.ModuleBuilderOpt.CurrentGenerationOrdinal;
             var containerName = GeneratedNames.MakeDynamicCallSiteContainerName(methodOrdinal, generation);
 
-            var synthesizedContainer = new DynamicSiteContainer(containerName, factory.TopLevelMethod);
+            var synthesizedContainer = new DynamicSiteContainer(containerName, factory.TopLevelMethod, factory.CurrentFunction);
             factory.AddNestedType(synthesizedContainer);
 
-            if (factory.TopLevelMethod.IsGenericMethod)
+            if (!synthesizedContainer.TypeParameters.IsEmpty)
             {
-                return synthesizedContainer.Construct(factory.TopLevelMethod.TypeParameters.Cast<TypeParameterSymbol, TypeSymbol>());
```

Comment:
I'm not sure why this cast was necessary...

Reply:
the cast is necessary in order to use the `ImmutableArray<TypeSymbol>` overload, otherwise we have to do an implicit conversion from `ImmutableArray<TypeParameterSymbol>` to `IEnumerable<TypeSymbol>` and call the other overload.

Reply:
Ah, that makes sense.